### PR TITLE
[MM-45688]: forward messages to new dm

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
this PR adds the feature to forward a message from a public channel to a new DM channel (a DM channel with a user that did not exist before).

#### Ticket Link
[MM-45688](https://mattermost.atlassian.net/browse/MM-45688)

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "requires": true,
   "packages": {
     "": {
+      "name": "mattermost-webapp",
```
Comment (AshishDhama) [inline package-lock.json:8]: @michelengelen package-lock.json tricked you again, @M-ZubairAhmed are we planning to add this?

Comment (michelengelen) [reply package-lock.json:8]: Argh! 😂

Comment (M-ZubairAhmed) [reply package-lock.json:8]: yeah lets keep it in this PR, it was removed from a PR from our own team 😛 missed that.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @michelengelen 
Tested, looks good to merge.
Able to forward posts to new DMs
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/239402-c94be7b-onprem-ent-webapp-pr-11036-mm-ee-test:c94be7b/mochawesome.html) looks good - no PR related failures

**Comment by michelengelen** [comment]
@AshishDhama or @koox00 kind reminder to re-review

**Review by AshishDhama — APPROVED** [review]

**Pull request merged**